examples/test_resnet_vitis_ai_ptq_cpu.py

```diff
@@ -0,0 +1,87 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import json
+import os
+from pathlib import Path
+
+import pytest
+
+from olive.common.utils import retry_func, run_subprocess
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup():
+    """setup any state specific to the execution of the given module."""
+    cur_dir = Path(__file__).resolve().parent
+    example_dir = cur_dir / "resnet_vitis_ai_ptq_cpu"
+    os.chdir(example_dir)
+
+    # prepare model and data
+    # retry since it fails randomly
+    retry_func(run_subprocess, kwargs={"cmd": "python prepare_model_data.py", "check": True})
+
+    yield
+    os.chdir(cur_dir)
+
+
+def check_output(footprints):
+    for footprint in footprints.values():
+        assert footprint.nodes is not None
+        for v in footprint.nodes.values():
+            assert all([value > 0 for value in v.metrics.value.values()])
+
+
+@pytest.mark.parametrize("search_algorithm", ["exhaustive"])
+@pytest.mark.parametrize("execution_order", ["pass-by-pass"])
+@pytest.mark.parametrize("system", ["local_system", "aml_system"])
+@pytest.mark.parametrize("olive_json", ["resnet_static_config.json"])
+def test_resnet(search_algorithm, execution_order, system, olive_json):
+    # TODO: add gpu e2e test
+    from olive.workflows import run as olive_run
+
+    olive_config = None
+    with open(olive_json, "r") as fin:
+        olive_config = json.load(fin)
+
+    # update search strategy
+    olive_config["engine"]["search_strategy"]["search_algorithm"] = search_algorithm
+    if search_algorithm == "random" or search_algorithm == "tpe":
+        olive_config["engine"]["search_strategy"]["search_algorithm_config"] = {"num_samples": 3, "seed": 0}
+    olive_config["engine"]["search_strategy"]["execution_order"] = execution_order
+
+    # set aml_system as dev
+    olive_config["systems"]["aml_system"]["config"]["is_dev"] = True
+
+    # update host and target
+    olive_config["engine"]["host"] = system
+    olive_config["engine"]["target"] = system
+
+    if system == "aml_system":
+        generate_olive_workspace_config("olive-workspace-config.json")
+
+    footprint = olive_run(olive_config)
+    check_output(footprint)
+
+
+def generate_olive_workspace_config(workspace_config_path):
```

Comment:
Please refer to the case in main branch to update way for workspace config generation.